src/protocols/login.js

```diff
@@ -278,30 +292,25 @@ function verifySignedPacketV01(nonceDB: NonceDB, origin: string, jwsHeader: JwsH
   const ksignAddr = ethUtil.pubToAddress(jwsPayload.ksign, true);
   const ksignAddrHex = utils.bytesToHex(ksignAddr);
   if (!utils.verifySignature(utils.bytesToHex(msgHash), sigHex, ksignAddrHex)) { // mHex, sigHex, addressHex
-    return undefined;
+    return false;
   }
 
   // 7. VerifyProofOfClaim(jwsPayload.proofOfKSign, relayPk)
   if (!proofs.verifyProofClaim(jwsPayload.proofKSign, relayAddr)) {
-    return undefined;
+    return false;
   }
 
-  switch (jwsPayload.type) {
-    case IDENASSERTV01:
-      return verifyIdenAssertV01(nonceDB, origin, jwsHeader, jwsPayload);
-    default:
-      return undefined;
-  }
+  return true;
 }
 
 /**
  * Verify a signed packet
  * @param {Object} nonceDB
  * @param {String} origin
  * @param {String} signedPacket
- * @returns {Object} nonce
+ * @returns {undefined | Object} deserialization output with verification result
  */
-function verifySignedPacket(nonceDB: NonceDB, origin: string, signedPacket: string): ?NonceVerified {
+function verifySignedPacket(signedPacket: string): ?{ verified: boolean, header: JwsHeader, payload: JwsPayload } {
```

Comment:
This could be generic function verification. Hence, we should define function like `verifyGenericPacket` for specific packet type `GENERICSIGV01`

Reply:
done